### ISSUE TITLE
PHP 7.3 Warning: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"? tcpdf.php on line 17785

### ISSUE THREAD

**Issue by webhostingsrbijaRS** [tecnickcom/TCPDF#188]
after i switch to 7.3 i have this warning.
couple minutes before posting here i updated TCPDF but error still showing 
custom script  implemented in wordpress

this warning don't break anything and pdf is created without error

**Comment by mvsyed** [comment]
Error 500
"continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?
 same problem

**Comment by serj-12** [comment]
Deprecate and remove continue targeting switch:
[https://wiki.php.net/rfc/continue_on_switch_deprecation](https://wiki.php.net/rfc/continue_on_switch_deprecation)

In PHP, if continue is applied to a switch statement, it behaves the same as break. In other languages, it would continue the surrounding loop instead. To avoid confusion, this RFC proposes to deprecate and remove continues acting on switch.

The following example demonstrates the current behavior:

```
while ($foo) {
    switch ($bar) {
        case "baz":
            continue; // In PHP: Behaves like "break;"
                      // In C:   Behaves like "continue 2;"
    }
}
```
While continue and break are synonymous in this context, break is used almost exclusively, by convention and due to familiarity from other languages. Attempts to use continue are almost certainly attempts to achieve the behavior of continue 2 by programmers who are not familiar with this PHP-ism.

**Proposal**
This RFC proposes to deprecate in PHP 7.3 and remove in PHP 8 the ability to target a switch statement through continue. This is no loss in functionality (it is always possible to replace the continue with a break), but avoids a gotcha for programmers coming from other languages.

The following code illustrates cases which are deprecated and by what they can be replaced at various levels of loop nesting:
```
switch ($foo) {
    case "bar":
        continue; // Deprecated
        break;    // Use this for equivalent behavior
}
 
while ($foo) {
    switch ($bar) {
        case "baz":
            continue;   // Deprecated
            break;      // Use this for equivalent behavior
            continue 2; // Or this for C-like behavior
    }
}
 
while ($foo) {
    switch ($bar) {
        case "baz":
            while ($xyz) {
                continue 2; // Deprecated
                break 2;    // Use this for equivalent behavior
                continue 3; // Or this for C-like behavior
            }
    }
}
```
**Unaffected PHP Functionality**
Continue can still be used inside switch statements, as long as it does not target the switch. The meaning of continue and break inside switch never changes, some cases are just forbidden.

To further clarify which uses of continue are affected, please consider the following example:

```
while ($foo) {
    switch ($bar) {
        case "baz":
            while ($xyz) {
                continue;   // Targeting the inner while loop: Allowed
                continue 2; // Targeting the switch: Deprecated
                continue 3; // Targeting the outer while loop: Allowed
            }
    }
}
```

In tcpdf.php:

```
case 're': {
    // justify block
   if (!TCPDF_STATIC::empty_string($this->lispacer)) {
	$this->lispacer = '';
	continue; // continue should be replaced with break OR continue 2
   }
.....
.....

```
I used both **break** and **continue 2** and in both cases I did not get that warning and the PDF files were created.

**Comment by webhostingsrbijaRS** [comment]
i fixed for my self but i posted here to get attention because not everyone know how to fix this issue, when admin see this and fix with new update we will not have this warning

**Comment by mvsyed** [comment]
Yes, It works fine, Thanks brother, here my site pdf files are generating without any issue.
https://freeseorank com

**Comment by nicolaasuni** [comment]
Thank you for reporting this. TCPDF 6 is now a legacy project and is officially **deprecated**, so no further fixes or features will be merged into this repository. As part of this transition, we are closing older issues. Please migrate to **[tc-lib-pdf](https://github.com/tecnickcom/tc-lib-pdf)** and, if the same problem still occurs there, open a new issue with updated reproduction steps. For more context, see the pinned announcement: [Thank You - and the Future of TCPDF](https://github.com/tecnickcom/TCPDF/issues/867).
